### ISSUE TITLE
Truncated file from MTP device due to loading directory listing during the file' creation

### ISSUE THREAD

**Issue by HT-7** [linuxmint/nemo#2958]
* Nemo version (nemo --version): 3.6.4
 * Is issue with desktop or windowed nemo? Desktop.
 * Linux Mint 18.3 (LTS) 64-Bit

----
### Issue
A file transferred from an MTP device shortly after creation could end up truncated as a result of the directory listing being loaded in the wrong moment, meaning before the file was complete, and the file manager assuming it was transferred completely while it wasn't.

That is because the listed size is below the actual size of the file when it is finished.

When moving out files, this could cause the source file to be deleted without having been properly transferred, i. e. data loss.

### Steps to reproduce
Create a large ZIP file of hundreds of megabytes using the smartphone's or tablet's file manager. The fact that it is a ZIP file does not matter, but its relatively slow creation due to the compression process gives time for the step below.

In Nemo, open the parent directory of that ZIP file through MTP (i. e. using a USB cable) **while** it is in the process of being created.

After the file is complete, copy or move it out. The file on the target location is possibly truncated, meaning incomplete.

This will presumably only work if the smartphone's file manager does not use a temporary file name during the ZIP file's creation that is different from the file name after it is completed. For example, Samsung's pre-installed file manager and RhythmSoft File Manager HD do not use a temporary name for the ZIP file. I have not tested it with ES File Explorer (pre-adware version) and Amaze File Manager yet, since ES does not retain files' and folders' last modified time stamps in ZIP archives, making me not use it for creating ZIP archives, and Amaze File Manager frequently crashes, making it not so useful.

### Expected behaviour
One could go about it by verifying the size of each file after transfer. I assume that it has a slight impact on performance, especially for many small files, however it might be worth it. The purpose of this post is just to inform that this problem exists.

If the file sizes do not match, a warning could be displayed, and offer the user to retry transferring. However, in order to save time after the user returns from absence from the computer, said warning should only be displayed **after** the rest of the selection has finished transferring instead of when the mismatch is detected.
In other words, _first_ get the simple work done and _then_ handle errors. Don't let errors delay the simple work. (Also applies to other errors. Could be a separate issue.)

**Comment by Jeremy7701** [comment]
If you try to process incomplete files, this is bound to happen.
The solution is to create (and possibly check) that the mobile device has finished creating its output files.
Since a ZIP file contains the list of the zipped files near the end of the file, the problem will be particularly acute for ZIP files.

**Comment by leigh123linux** [comment]
Linux Mint 18.3 isn't supported, you need to update.

https://endoflife.software/operating-systems/linux/linux-mint

If you can reproduce with nemo-5.2.x feel free to reopen.
